examples/chat/src/routes/messages.rs

```diff
@@ -71,11 +72,13 @@ pub async fn create(request: Request, _: Next) -> via::Result {
         insert.get_result(&mut conn).await?
     };
 
-    // Notify subscribers that a message has been created.
-    head.state()
-        .publish(current_user_id, thread_id, Event::Message(&message))?;
+    let context = EventContext::new(Some(thread_id), current_user_id);
+    let event = Event::Message(message);
 
-    Response::build().status(201).json(&message)
+    // Notify subscribers that a message has been created and respond.
+    head.state()
+        .publish(context, event)?
+        .finalize(Response::build().status(201))
```

Comment:
I'm really happy with how the `Finalize` trait composes with a `ResponseBuilder`. This solves a rather complex problem in a pretty simple way. A trend that we want to continue with this framework.